src/lib/core/components/visualizations/VisualizationsContainer.tsx

```diff
@@ -275,6 +273,20 @@ function FullScreenVisualization(props: Props & { id: string }) {
     [updateVisualizations, id]
   );
 
+  // update currentPlotfilters with the latest filters at fullscreen mode
+  useEffect(() => {
+    updateVisualizations((visualizations) =>
+      visualizations.map((v) =>
+        v.visualizationId !== id
+          ? v
+          : {
+              ...v,
+              descriptor: { ...v.descriptor, currentPlotFilters: filters },
+            }
+      )
+    );
+  }, [filters]);
```

Comment:
This effect has a missing `updateVisualizations` dependency, which is a [potential source of bugs](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies). Since a stale value of `updateVisualizations` could cause a user's analyses to be mis-saved, we should address.

As you probably discovered on your own, we can't just add `updateVisualizations` to the effect's dependency list, because this causes an infinite loop! (The effect changes the `visualizations`, which causes `updateVisualizations` to change, which causes `visualizations` to change, which causes `updateVisualizations` to change...)

The React docs have a few [solutions](https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often) for this issue, but only one immediately works here: making a ref to the latest version of `updateVisualizations` and using that ref in the effect.

```suggestion
  // store a ref to the latest version of updateVisualizations
  const updateVisualizationsRef = useRef(updateVisualizations);
  
  // whenever updateVisualizations changes, update the updateVisualizationsRef
  useEffect(() => {
    updateVisualizationsRef.current = updateVisualizations;
  }, [updateVisualizations]);
  
  // update currentPlotFilters with the latest filters at fullscreen mode
  useEffect(() => {
    updateVisualizationsRef.current((visualizations) =>
      visualizations.map((v) =>
        v.visualizationId !== id
          ? v
          : {
              ...v,
              descriptor: { ...v.descriptor, currentPlotFilters: filters },
            }
      )
    );
  }, [filters]);
```

Reply:
@jtlong3rd Thank you for your detailed explanation! 👍 Honestly I didn't know about the useRef solution to resolve this kind of dependency issue, so it is great to learn about that! I have committed with your solution.

Reply:
Also thanks for the education.
Presumably `updateVisualizationsRef` doesn't need to be in the dependencies.  Like `setState` functions from `useState` - they are guaranteed not to change?

Reply:
Yup, it's OK to omit `updateVisualizationsRef` from the effect dependencies, because it's just a referentially stable container. (As with `useState` setters, our linter is smart enough to permit that omission with out complaining.)

Also of note - you should also not add `updateVisualizatuionsRef.current` to the dependencies. If you do that, our linter will issue a warning. The reason is that `updateVisualizatuionsRef.current` is a mutable value which isn't being tracked by React, and so is not a valid effect dependency.

(Great blog post on this topic [here](https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/).)